iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/bucket/projectorg/DocumentOrganizationCombinerTest.java

```diff
@@ -0,0 +1,193 @@
+        // given
+        stub();
+        Float docProjConfidenceLevelThreshold = 0.5f;
+        // execute
+        JavaRDD<AffMatchDocumentOrganization> result = builder.build(documentProjectRdd, projectOrganizationRdd,
+                docProjConfidenceLevelThreshold);
+        // assert        
+        assertResults(result, docProjConfidenceLevelThreshold);
+    }
+
+    // ------------------------ PRIVATE --------------------------
+
+    private void stub() {
+        doReturn(documentProjectPairedRdd).when(documentProjectRdd).keyBy(any());
+        doReturn(documentProjectPairedFilteredRdd).when(documentProjectPairedRdd).filter(any());
+        doReturn(projectOrganizationPairedRdd).when(projectOrganizationRdd).keyBy(any());
+        doReturn(documentOrganizationJoinedRdd).when(documentProjectPairedFilteredRdd).join(any());
+        doReturn(documentOrganizationMappedRdd).when(documentOrganizationJoinedRdd).map(any());
+        when(documentOrganizationMappedRdd.distinct()).thenReturn(documentOrganizationDistinctRdd);
+    }
+
+    private void assertResults(JavaRDD<AffMatchDocumentOrganization> result, Float docProjConfidenceLevelThreshold)
+            throws Exception {
+        assertTrue(result == documentOrganizationDistinctRdd);
+        verify(documentProjectRdd).keyBy(keyByDocumentProjectFunction.capture());
+        assertKeyByDocumentProjectFunction(keyByDocumentProjectFunction.getValue());
+        verify(documentProjectPairedRdd).filter(filterDocumentProjectFunction.capture());
+        assertFilterDocumentProjectFunction(filterDocumentProjectFunction.getValue(),
+                docProjConfidenceLevelThreshold);
+        verify(projectOrganizationRdd).keyBy(keyByProjectOrganizationFunction.capture());
+        assertKeyByProjectOrganizationFunction(keyByProjectOrganizationFunction.getValue());
+        verify(documentProjectPairedFilteredRdd).join(projectOrganizationPairedRdd);
+        verify(documentOrganizationJoinedRdd).map(mapDocProjJoinedWithProjOrgToDocOrgFunction.capture());
+        assertMapDocumentOrganizationFunction(mapDocProjJoinedWithProjOrgToDocOrgFunction.getValue());
+        verify(documentOrganizationMappedRdd).distinct();
+    }
+
+    private void assertKeyByDocumentProjectFunction(Function<AffMatchDocumentProject, String> function)
+            throws Exception {
+        // given
+        AffMatchDocumentProject docProj = new AffMatchDocumentProject(documentId, projectId, confidenceLevel);
+        // execute
+        String projId = function.call(docProj);
+        // assert
+        assertThat(projId, equalTo(projectId));
+    }
+
+    private void assertFilterDocumentProjectFunction(Function<Tuple2<String, AffMatchDocumentProject>, Boolean> function,
+            Float confidenceLevelThreshold) throws Exception {
+        assertTrue(function.call(
+                new Tuple2<String, AffMatchDocumentProject>(projectId, new AffMatchDocumentProject(documentId, projectId, 0.6f))));
```

Comment:
Well, what if the passed confidenceLevel will be greater that 0.6f? Maybe the argument to this function should be confidenceLevelThreshold!=null?confidenceLevelThreshold+ 0.1f:0.1f instead of 0.6f?